### PR TITLE
B: fix a bunch of SCA warnings

### PR DESCRIPTION
**Pull request by MrPepperoni**

### CODE REVIEW

File: src/core/damage_taken_event.h [hunk 1]
```diff
@@ -14,8 +14,8 @@ struct DamageTakenEvent : public Event
     int32_t Damage;
     Type type;
     int32_t ActorGUID;
-    DamageTakenEvent( double x, double y ): Pos( x, y ), type( Health ) {}
-    DamageTakenEvent( glm::vec2 const& p ): Pos( p ), type( Health ) {}
```
Comment (abannerth) [inline src/core/damage_taken_event.h:18]: How about using a delegating constructor?

Comment (MrPepperoni) [inline src/core/damage_taken_event.h:18]: good point, will do

File: src/render/recognizer_repo.cpp [hunk 2]
```diff
@@ -35,18 +35,29 @@ bool RecognizerRepo::AddRecognizerFromOneDesc( Json::Value& RecognizerDesc )
...
+            continue;
         }
         recognizer->SetActionRenderer( AutoId( rendererName ) );
         recognizer->SetOrder( orderindex++ );
-        recognizers.push_back( recognizer.release() );
+        try
+        {
+            recognizers.push_back( recognizer.release() );
+        }
+        catch( boost::bad_pointer const& )
```
Comment (weanti) [inline src/render/recognizer_repo.cpp]: can it be a bad_ptr if recognizer is checked for nullprt before?

Comment (MrPepperoni) [inline src/render/recognizer_repo.cpp]: i think not. only saw the missing nullptr check afterwards.
got to check the boost sources regardless.

Comment (weanti) [inline src/render/recognizer_repo.cpp]: It just checks for 0 == x (in enforce_null_policy), so try catch is not necessary

**Comment by MrPepperoni** [comment]
review notes fix't @ 79bce0a

File: src/render/recognizer_repo.cpp (hunk 2 above)
Comment (MrPepperoni) [inline src/render/recognizer_repo.cpp]: apparently Coverity isn't aware of this check, it is only concerned about the possibility of a throw()
( which actually makes sense, the implementation of the third party might change to also throw in some other instances )

**Pull request merged**